src/app/login/page.tsx

```diff
@@ -15,11 +21,50 @@ const LoginPage = () => {
   });
 
   const [errorMessage, setError] = useState('');
+  const [emailError, setEmailError] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
 
   const router = useRouter();
 
   const { login } = useAuth();
 
+  const validateEmail = () => {
+    if (
+      !formData.email ||
+      !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(formData.email)
+    ) {
+      setEmailError('Zadejte prosím validní email.');
+
+      return false;
+    }
+    setEmailError('');
+    setError('');
+
+    return true;
+  };
```

Comment:
This validation, the same as the password should be abstracted and moved into the input itself onBlur. We would only pass the different validation functions inside. The input itself would then handle showing the error itself - alongside the proper accessibility for it. This would allow us to get rid of all of most of the validation states

BUT! for now, this can be implemented via the native HTML 5 fields on the input like pattern, required, min/max -> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#client-side_validation

I also do believe, that a native email pattern is enough for the email, so `type="email"` will suffice. We will implement hook-form alongside yup latter

Reply:
Those html validations will actually disable submitting I believe until everything is fixed